skglm/solvers/prox_newton_solver.py

```diff
@@ -0,0 +1,312 @@
+
+    Returns
+    -------
+    w : array, shape (n_features,)
+        Coefficient vector.
+
+    obj_out : array, shape (n_iter,)
+        The objective values at every outer iteration.
+
+    stop_crit : float
+        Value of stopping criterion at convergence.
+    """
+    if ws_strategy not in ("subdiff", "fixpoint"):
+        raise ValueError(f"Unsupported value for ws_strategy: {ws_strategy}")
+    if not isinstance(datafit, (Logistic, Logistic_32)):
+        raise ValueError("Prox-Newton solver only supports Logistic datafits.")
+    n_features = X.shape[1]
+    if n_features >= 10_000:
+        warnings.warn(
+            "Prox-Newton solver can be prohibitively slow for high-dimensional " +
+            "problems. We recommend using `cd_solver` instead for faster convergence")
+    pen = penalty.is_penalized(n_features)
+    unpen = ~pen
+    n_unpen = unpen.sum()
+    obj_out = []
+    all_feats = np.arange(n_features)
+    stop_crit = np.inf  # initialize for case n_iter=0
+    accelerator = AndersonAcceleration(K=5)
+
+    is_sparse = sparse.issparse(X)
+    for t in range(max_iter):
+        if is_sparse:
+            grad = datafit.full_grad_sparse(
+                X.data, X.indptr, X.indices, y, Xw)
+        else:
+            grad = construct_grad(X, y, w, Xw, datafit, all_feats)
+
+        if ws_strategy == "subdiff":
+            opt = penalty.subdiff_distance(w, grad, all_feats)
+        elif ws_strategy == "fixpoint":
+            opt = dist_fix_point(w, grad, datafit, penalty, all_feats)
+        stop_crit = np.max(opt)
+        if verbose:
+            print(f"Stopping criterion max violation: {stop_crit:.2e}")
+        if stop_crit <= tol:
+            break
+        # 1) select features: all unpenalized + 2 * (nnz and penalized)
+        ws_size = max(min(p0 + n_unpen, n_features),
+                      min(2 * penalty.generalized_support(w).sum() -
+                          n_unpen, n_features))
+
+        opt[unpen] = np.inf  # always include unpenalized features
+        opt[penalty.generalized_support(w)] = np.inf
+
+        # taking arg_top_K features violating the stopping criterion
+        ws = np.argpartition(opt, -ws_size)[-ws_size:]
+
+        if verbose:
+            print(f"Iteration {t + 1}, {ws_size} feats in subpb.")
+
+        # 2) run prox newton on smaller subproblem
+        for epoch in range(max_epochs):
+            _max_pn_cd_epochs = 1 if epoch == 0 else max_pn_cd_epochs
+            pn_tol = 0 if epoch == 0 else tol
+            if is_sparse:
+                _prox_newton_iter_sparse(
+                    X.data, X.indptr, X.indices, Xw, w, y, penalty, ws,
+                    min_pn_cd_epochs, _max_pn_cd_epochs, max_backtrack, pn_tol)
+            else:
+                _prox_newton_iter(
+                    X, Xw, w, y, penalty, ws, min_pn_cd_epochs, _max_pn_cd_epochs,
+                    max_backtrack, pn_tol)
+
+            # 3) do Anderson acceleration on smaller problem
```

Comment:
You can remove this part, in my experience it's not useful to do AA for prox newton